.github/workflows/ci.yml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+
+env:
+  IMAGE_TAG: ${{ github.sha }}
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - name: Build Docker images
+        run: |
+          TARGET=prod make build
+          TARGET=dev make build
+
+      - name: Save Docker images
+        run: |
+          mkdir -p docker-image-prod docker-image-dev
+          docker save -o docker-image-prod/article-page.tar libero/article-page:${IMAGE_TAG}
+          gzip docker-image-prod/article-page.tar
+          docker save -o docker-image-dev/article-page.tar libero/article-page:${IMAGE_TAG}-dev
+          gzip docker-image-dev/article-page.tar
+
```

Comment:
Is the gzip necessary? To my understanding github zips all artifacts by default.

Reply:
https://github.com/libero/article-store/pull/74/commits/0618c941b624d0defb1b3e8737c7e1e8783e6f86 introduced it. Comparing that build (https://github.com/libero/article-store/runs/334735733) and the previous one (https://github.com/libero/article-store/runs/334670782) the artifacts are a lot smaller.

Unless it's a new feature, is needed.

Reply:
Though reading https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts#about-workflow-artifacts, they up/downloaded in a ZIP anyway, and that size is just what's inside?

Reply:
Let's take it out, and compare the build log.

Reply:
Ah, forgot there's no build on this PR...

Reply:
How do I get a build running? https://github.com/libero/article-store/pull/74 had them... 🤔

Reply:
Is it possible the workflow has to be in `master`? Or, this being open from a fork may only authorize it to run in the fork?

Reply:
> the artifacts are a lot smaller

Space is cheap/free; what we compared/should compare now is the upload and download time (really the execution time of those steps) which IIRC improved at the time; we should measure again now.

Reply:
> Is it possible the workflow has to be in `master`? Or, this being open from a fork may only authorize it to run in the fork?

Your one ran from a fork too...

Reply:
Looking into [the parent commit the PR was forked from](https://github.com/libero/article-store/tree/2e43f287e3c0887607a21295156c5fc94957df70) shows no existing Github Actions. And nothing related got in between the opening of the PR and its merge.

Reply:
Next thing would be to check https://github.com/thewilkybarkid/article-page/settings/actions (hand-written URL) to see if something needs to be enabled. https://github.com/libero/article-page/settings/actions looks fine.

Reply:
Might be that https://github.com/libero/article-store/pull/71 was enough for GitHub actions to recognise the repo and start working.

Reply:
https://github.com/libero/article-page/pull/3 was recognised. Will add a new commit here.

Reply:
https://github.com/libero/article-page/pull/1/commits/5f3034237dcfada6a4a796b85e3bbad4cc72481f built.

Reply:
All sorted.

Going back to:

> Is the gzip necessary? To my understanding github zips all artifacts by default.

We've decided to leave off the local compression, and will test it out separately and then introduce it if needed.